core/src/main/java/com/datastrato/gravitino/authorization/AdminManager.java

```diff
@@ -37,37 +37,54 @@ class AdminManager {
   private final EntityStore store;
   private final IdGenerator idGenerator;
   private final List<String> serviceAdmins;
+  private final RoleManager roleManager;
 
-  AdminManager(EntityStore store, IdGenerator idGenerator, Config config) {
+  AdminManager(EntityStore store, IdGenerator idGenerator, Config config, RoleManager roleManager) {
     this.store = store;
     this.idGenerator = idGenerator;
     this.serviceAdmins = config.get(Configs.SERVICE_ADMINS);
+    this.roleManager = roleManager;
+
+    updateSystemMetalakeUsers();
   }
 
   User addMetalakeAdmin(String user) {
-
-    UserEntity userEntity =
-        UserEntity.builder()
-            .withId(idGenerator.nextId())
-            .withName(user)
-            .withNamespace(
-                Namespace.of(
-                    Entity.SYSTEM_METALAKE_RESERVED_NAME,
-                    Entity.AUTHORIZATION_CATALOG_NAME,
-                    Entity.ADMIN_SCHEMA_NAME))
-            .withRoleNames(Lists.newArrayList())
-            .withAuditInfo(
-                AuditInfo.builder()
-                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
-                    .withCreateTime(Instant.now())
-                    .build())
-            .build();
     try {
-      store.put(userEntity, false /* overwritten */);
-      return userEntity;
+      if (isServiceAdmin(user)) {
```

Comment:
Since the user is a service admin, do we still need to add roles `SYSTEM_METALAKE_MANAGE_USER_ROLE` and `METALAKE_CREATE_ROLE` to it?

Reply:
Yes, because we used the role to execute authorization operation.